### PR TITLE
fix(ui): Improve handling of errors in Form components

### PR DESCRIPTION
**Pull request by dcramer**
- Cache error in addition to value
- Use getValue helper for value computation

### CODE REVIEW

File: src/sentry/static/sentry/app/components/forms/formField.jsx [hunk 1]
```diff
@@ -38,18 +38,21 @@ export default class FormField extends React.PureComponent {
...
 
   componentWillReceiveProps(nextProps, nextContext) {
-    if (
-      this.props.value !== nextProps.value ||
-      (!defined(this.context.form) && defined(nextContext.form))
-    ) {
-      this.setValue(this.getValue(nextProps, nextContext));
+    let newError = this.getError(nextProps, nextContext);
+    let newValue = this.getValue(nextProps, nextContext);
+    if (newError != this.state.error) {
```
Comment (dcramer) [inline src/sentry/static/sentry/app/components/forms/formField.jsx]: there's two setState calls here, which kind of is meh, but I **think** it'll be fine

Comment (billyvg) [reply src/sentry/static/sentry/app/components/forms/formField.jsx]: It's meh, but it's fine, react batches setState calls.

**Comment by dcramer** [comment]
@campbellita

**Review by billyvg — APPROVED** [review]
My guess is that formfield is a PureComponent, and when we update `error` in Form, it compares state shallowly and doesn't update. But this works so

**Comment by dcramer** [comment]
@billyvg oh you're right it is -- I could revert this and take that out, but this almost seems like a better solution?

**Comment by billyvg** [comment]
Yeah let's keep it.

**Comment by dcramer** [comment]
i give up, fu selenium

**Pull request merged**